peakina/readers/csv.py

```diff
@@ -51,23 +51,24 @@ def read_csv(
     )
 
 
-def _line_count(filepath_or_buffer: "FilePathOrBuffer") -> int:
-    with open(filepath_or_buffer) as f:
-        lines = 0
-        buf_size = 1024 * 1024
-        read_f = f.read  # loop optimization
+def csv_meta(
+    filepath_or_buffer: "FilePathOrBuffer", reader_kwrgs: Dict[str, Any]
+) -> Dict[str, Any]:
+
+    preview_nrows, preview_offset = None, None
+    if "preview_offset" in reader_kwrgs:
+        preview_offset = reader_kwrgs.pop("preview_offset")
 
-        buf = read_f(buf_size)
-        while buf:
-            lines += buf.count("\n")
-            buf = read_f(buf_size)
+    if "preview_nrows" in reader_kwrgs:
+        preview_nrows = reader_kwrgs.pop("preview_nrows")
 
-        return lines
+    df_rows = pd.read_csv(
+        filepath_or_buffer,
+        **reader_kwrgs,
+        skiprows=range(1, preview_offset + 1) if preview_offset else None,
+        nrows=preview_nrows,
+    ).shape[0]
 
+    total_rows = pd.read_csv(filepath_or_buffer, **reader_kwrgs).shape[0]
```

Comment:
`_line_count` has been created to count very fast the total of rows of a huge csv!
We don't want to load a full CSV just to get its count

Reply:
I would have done something like
```py
total_rows = _line_count(filepath_or_buffer)

preview_offset = reader_kwargs.pop("preview_offset", 0)
preview_nrows = reader_kwargs.pop("preview_nrows", None)
if preview_nrows is not None:
    df_rows = min(total_rows, preview_nrows - preview_offset)
else:
    df_rows = total_rows
```
with a test
- where we ask all the rows of a file of 4 rows and `df_rows` must equal 4
- where we ask 5 rows of a file of 4 rows and `df_rows` must equal 4
- where we ask 3 rows of a file of 4 rows and `df_rows` must equal 3

Wouldn't that work?

Reply:
thanks for updates, i replicates these tests scenarios here : https://github.com/ToucanToco/peakina/pull/77/commits/5d34fe5c7d30125798e8638c3d1e630519a9604c